### ISSUE TITLE
IP address wrongly captured

### ISSUE THREAD

**Issue by shaman1307** [zaneclaes/network-traffic-metrics#3]
When I run py script as
sudo python3 /home/pi/network-traffic-metrics/network-traffic-metrics.py "(src net 192.168.8.0/24 and not dst net 192.168.8.0/24) or (dst net 192.168.8.0/24 and not src net 192.168.8.0/24)"

I see here http://192.168.8.1:8000/metrics
the next stranger IP. Mostly for ICMP proto

ntm_packets_total{dst="168.8",proto="icmp",service="",src="cust.tele2"} 8.0
ntm_packets_total{dst="168.8",proto="icmp",service="",src="59.49"} 1.0
ntm_packets_total{dst="168.8",proto="icmp",service="",src="6.3sbsdatacom-pvt-ltd"} 1.0
ntm_packets_total{dst="168.8",proto="icmp",service="",src="static-31-148-113-93.netbynet"} 2.0
ntm_packets_total{dst="168.8",proto="icmp",service="",src="108.216"} 1.0
ntm_packets_total{dst="igmp.mcast",proto="igmp",service="",src="168.8"} 11319.0

And I suppose because of such script behavior, not all traffic is reflected in Grafana

**Comment by zaneclaes** [comment]
Two possibilities:
1) Your IP addresses do not match the regex `[\w\d\.-]+`, and therefore are not valid IPs
2)`tcpdump` is not reporting the full IP address.

I'd try running `tcpdump` directly (using the same parameters) and examining the output to see why it's not matching. If you want to post the output here I can take a look.

**Comment by shaman1307** [comment]
I found tcpdump with icmp protocol that could be parsed as "168.8"
Those lines looks like:
20:04:07.887449 IP (tos 0xc0, ttl 64, id 6849, offset 0, flags [none], proto ICMP (1), length 157)
    46.175.232.60.maxnet.net.pl > 192.168.8.101: ICMP host 46.175.232.60.maxnet.net.pl unreachable - admin prohibited, length 137
	IP (tos 0x0, ttl 64, id 33875, offset 0, flags [DF], proto UDP (17), length 129)
    192.168.8.101.6881 > 46.175.232.60.maxnet.net.pl.6881: UDP, length 101[|icmp]

I'm not sure that I need sush info in the report, because all of them contains "unreachable" word

**Comment by zaneclaes** [comment]
Yeah, that sounds right. In particular, I think it's the `- admin prohibited` message that breaks it. I've never seen a message like that before. I'm sure we could fix the regex, but I'm wondering if this behavior is expected. Is there some sort of firewall or something causing this?

**Comment by shaman1307** [comment]
On my Rspi 4 only standard firewall was enabled. As per your instructions

**Comment by zaneclaes** [comment]
Hm. [everything I can find](https://unixspace.wordpress.com/2013/06/12/35/comment-page-1/) says that message is caused by a firewall. Keep in mind, firewalls don't only live on your Raspberry Pi. If there's anything upstream of it, the firewall might live there. Do you recognize the `maxnet.net.pl` URL? It looks like it may be a Polish ISP, in which case I wonder if this is on their end.

**Comment by shaman1307** [comment]
maxnet.net.pl is my Internet provider

**Comment by zaneclaes** [comment]
Okay, here's what I'm seeing then. The line which is breaking things is:

```
46.175.232.60.maxnet.net.pl > 192.168.8.101: ICMP host 46.175.232.60.maxnet.net.pl unreachable - admin prohibited, length 137
IP (tos 0x0, ttl 64, id 33875, offset 0, flags [DF], proto UDP (17), length 129)
```

(Okay, two lines). This indicates that the ISP (maxnet) is trying to communicate with your device at `192.168.8.101`, but is being prevented from doing so. Which sounds very much like an ingress firewall to me, and may indicate packet loss in the communication between you and your ISP. 

In the link from the prior post, he suggested running `systemctl stop firewalld.service`. This is a good way to test for a short period of time and see if the error goes away. If it does, great, now find the config problem. If not, I'd try making 100% certain you don't have a different firewall installed, as well. For example, [the official Raspberry Pi docs recommend UFW](https://www.raspberrypi.org/documentation/configuration/security.md), which is a different firewall and would conflict with `firewalld` (which I had recommended). Note that there is no real "standard" firewall.

**Comment by zaneclaes** [comment]
Btw, if you need a work-around for now, you could either (1) adjust the script parameters to filter the offending IPs/URLs or (2) filter them after the fact in Grafana.

**Comment by nicolube** [comment]
@zaneclaes Pls prevent tcpdump from resolving ip addresses by adding the -n argungent.

https://wiki.ubuntuusers.de/tcpdump/

**Comment by zaneclaes** [comment]
@nicolube this would be a breaking change, insofar as that it is incompatible with the FQDN option. To support this, we'd need to resolve the resultant IP addresses via a DNS lookup. Ultimately this would be a much better solution than the current implementation, but it would require some thought & testing to accomplish properly. In particular, the latency introduced by the DNS lookup could create a critical bottleneck that slows the entire service in an unacceptable manner. If you have the time to implement & test such a change, I would more than welcome a PR.

**Comment by zaneclaes** [comment]
Closing due to inactivity.
